groupsforsections.py

```diff
@@ -0,0 +1,85 @@
+import json
+import sys
+import logging
+import requests
+import urllib
+import utils
+import yaml
+
+
+def create_group_category():
+    logging.debug('create_group_category() Called')
+
+    url = canvas_url + '/api/v1/courses/' + course_id + '/group_categories?name=' + urllib.quote_plus(
+            group_category_name)
+    response = api_handler(url, 'Post')
+    if response is None:
+        logging.error('Api call for \'creating group categories\' is not successful')
+        sys.exit(1)
```

Comment:
The exception here that is caught is due to for un know reason like socket timeout exception. So the Canvas won't send any error api  response object. when an exception occurs it will show the stack trace.  There should be enough details. Creating a group category is the first step towards the script and all the groups are created underneath that category and when that fails i don't want to script to  run any further. Currently this script is run on request basis and probably will run 1 time for each semester for each course with large sections only